### PR TITLE
Signup enhancement

### LINKED ISSUE #979: Updates to Sign Up Process
From the current UI: 
Screen Shot 2021-04-13 at 12 14 52 PM

1. Change the account selector to dropdown. 
--> Individual changes to Person; and is default.

2. Retrieve First & Last Name from Auth0 ID Token, and pre fill it when it is available.

3. For Reforestation Organizations, prefill address from config api, when available.
Screen Shot 2021-04-13 at 12 20 34 PM

### PR DESCRIPTION
**Pull request by rishsingh07**
Fixes #979
- [x] Change the account selector to dropdown.
--> Individual changes to Person; and is default.

- [x] Retrieve First & Last Name from Auth0 ID Token, and pre fill it when it is available.

- [x] For Reforestation Organizations, prefill address from config api, when available.

### CODE REVIEW

**Comment by sagararyal** [comment]
Please resolve the errors
- Crash on Reforestation Org due to loc.
- Layout issue on breakpoint < ~500px.

This was tested on Safari on macOS
https://user-images.githubusercontent.com/5401575/115234497-6115b900-a119-11eb-9f5d-d400ad65ff2f.mov

File: src/features/user/CompleteSignup/index.tsx [hunk 1]
```diff
@@ -74,7 +76,7 @@ export default function CompleteSignup() {
     setSnackbarOpen(false);
   };
 
-  const [type, setAccountType] = useState('individual');
+  const [type, setAccountType] = useState('person');
```
Comment (norbertschuler) [inline src/features/user/CompleteSignup/index.tsx:79]: Why don't you just change the translation from `Individual` to `Person` and leave the rest unchanged? I think you can not change the value for the `type` state as it is also used to communicate the user type into the backend and your change probably will break the API call.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by sarveshpro** [comment]
Error in complete signup
signup type - email signup
also remove token from console

Individual
image

TPO
image

File: src/features/user/CompleteSignup/index.tsx [hunk 2]
```diff
@@ -26,7 +27,7 @@ export default function CompleteSignup() {
     loginWithRedirect,
     user
   } = useAuth0();  
-
+console.log(user,"h")
```
Comment (norbertschuler) [inline src/features/user/CompleteSignup/index.tsx:30]: Please remove this before we merge the pull request.

File: src/features/user/CompleteSignup/index.tsx [hunk 3]
```diff
@@ -35,7 +36,7 @@ export default function CompleteSignup() {
   const isPrivate = watch('isPrivate');
 
   const [token, setToken] = React.useState('')
-
+console.log(token,"token")
```
Comment (norbertschuler) [inline src/features/user/CompleteSignup/index.tsx:39]: Please remove this before we merge the pull request.

File: src/features/user/CompleteSignup/index.tsx [hunk 4]
```diff
@@ -321,7 +332,10 @@ export default function CompleteSignup() {
               label={t('donate:country')}
               name="country"
               onChange={(country) => setCountry(country)}
-              defaultValue={defaultCountry}
+              defaultValue={getStoredConfig("loc").postalCode === "T1" || 
+                            getStoredConfig("loc").postalCode === "XX" || 
+                            getStoredConfig("loc").postalCode === "" ?
+                            "" : getStoredConfig("loc").postalCode}
```
Comment (norbertschuler) [inline src/features/user/CompleteSignup/index.tsx:338]: The registration currently fails for me saying no country is specified. Maybe this change for the country field is just a bug?

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by norbertschuler** [comment]
- [ ] On mobile devices the layout has some empty space between the bottom button and the bottom navigation when I test with a simulated iPhone X layout on Chrome on MacOS, but that's probably not related to this pull request:

Bildschirmfoto 2021-04-28 um 11 05 45

**Comment by rishsingh07** [comment]
@norbertschuler can you please check the signing up process if you're able to do it.

File: src/features/user/CompleteSignup/index.tsx (hunk 4 above)
Comment (norbertschuler) [reply src/features/user/CompleteSignup/index.tsx]: I now get an error trying to signup (on staging):

Bildschirmfoto 2021-04-29 um 09 42 18

**Comment by norbertschuler** [comment]
> @norbertschuler can you please check the signing up process if you're able to do it.

I now get an error signing up (on staging):

Bildschirmfoto 2021-04-29 um 09 42 18

**Comment by rishsingh07** [comment]
The issue was, the config was not present for staging. Sarvesh made it.

**Review by norbertschuler — APPROVED** [review]
Apart from the [layout bug](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/988#issuecomment-828296192) which is probably not caused by this pull request, I think this patch works as intented.

**Pull request merged**